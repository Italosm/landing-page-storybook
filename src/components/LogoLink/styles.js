import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    text-decoration: none;
    align-items: center;
    color: inherit;
    > img {
      max-height: 2.5rem;
    }
  `}
`;
