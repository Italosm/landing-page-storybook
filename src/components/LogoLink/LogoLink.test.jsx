import { screen } from '@testing-library/react';
import { LogoLink } from '.';
import { theme } from '../../styles/theme';
import { renderTheme } from '../../styles/render-theme';

describe('<LogoLink />', () => {
  it('should render text logo', () => {
    renderTheme(<LogoLink link="#target" text="The text" />);
    const textLogo = screen.getByRole('link', { name: 'The text' });
    expect(textLogo).toHaveAttribute('href', '#target');
    const heading = screen.getByRole('heading', { name: 'The text' });
    expect(heading).toBeInTheDocument();
  });

  it('should render img logo', () => {
    renderTheme(<LogoLink link="#target" text="The text" srcImg="img.jpg" />);
    const logo = screen.getByAltText('The text');
    expect(logo).toHaveAttribute('src', 'img.jpg');
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(
      <LogoLink link="#target" text="The text" srcImg="img.jpg" />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
