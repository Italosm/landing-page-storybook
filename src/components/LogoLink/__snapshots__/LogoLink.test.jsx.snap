// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogoLink /> should match snapshot 1`] = `
.c1 {
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: inherit;
}

.c1 > img {
  height: 3rem;
}

.c0 {
  color: #0A1128;
  font-size: 2.4rem;
  text-transform: uppercase;
}

<h1
  class="c0"
>
  <a
    class="c1"
    href="#target"
  >
    <img
      alt="The text"
      src="img.jpg"
    />
  </a>
</h1>
`;
